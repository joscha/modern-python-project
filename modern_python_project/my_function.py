def my_function(name: str = "World"):
    return f"Hello, {name}!"
