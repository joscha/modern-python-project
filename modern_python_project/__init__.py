from .my_function import my_function

__all__ = ["my_function"]
