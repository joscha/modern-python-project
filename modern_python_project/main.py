import click

from .my_function import my_function


@click.command()
@click.option("--name", prompt="Your name", help="The person to greet.")
def main(name):
    print(my_function(name=name))


if __name__ == "__main__":
    main()
