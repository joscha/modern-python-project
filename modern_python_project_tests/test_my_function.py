import logging

import pytest

from modern_python_project.my_function import my_function

logger = logging.getLogger(__name__)
logging.basicConfig(level=logging.INFO)


@pytest.mark.asyncio
async def test_async_function(snapshot):
    assert snapshot == my_function()
