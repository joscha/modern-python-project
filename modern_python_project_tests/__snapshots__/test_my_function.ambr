# serializer version: 1
# name: test_async_function
  'Hello, World!'
# ---
